### ISSUE TITLE
Some text is squished in app due to letter-spacing

### ISSUE THREAD

**Issue by msvivianso** [telus/tds-core#1233]
## Description

The `letter-spacing: -0.8px;` styling is affecting the text in the app to look squished. We inspected the affect elements and it looks like this styling is coming from `@tds/core-css-reset`

After talking with @agorovyi, he was able to reproduce the issue in the starter kit. He mentioned that it's happening due to `<span>` being inside `Heading` component.

Reference to slack thread about this issue: https://telusdigital.slack.com/archives/C66LP1WGG/p1565802218147600

## Reproduction Steps

1. Add a `<span>` div with a text inside the `Heading` component
2. Load a page with the above styling
3. See squished text inside the span

This styling is also in the `Text` component.

## Meta

- TDS component version: @tds/core-heading
- Willing to develop solution: No
- High impact: No

## Screenshots

**Expected** (`letter-spacing: 0px;`)
<img width="1680" alt="Screen Shot 2019-08-14 at 11 00 09 AM" src="https://user-images.githubusercontent.com/6042193/63055301-97665400-beb3-11e9-86ef-5b1129d8b0dd.png">

**Actual** (`letter-spacing: -0.8px;`)
<img width="1680" alt="Screen Shot 2019-08-14 at 11 00 02 AM" src="https://user-images.githubusercontent.com/6042193/63055337-a947f700-beb3-11e9-9431-38caaca2d3a0.png">

**Comment by theetrain** [comment]
Thanks for reporting. We're looking into a couple potential solutions:

### 1) Apply `letterSpacing: inherit` to `span`s within Heading and Typography components

By adding this selector to core-heading and other components, we can fix the edge case of nested `span`s not inheriting the `heading`'s letter spacing.

**Heading.jsx**
```diff
  h1: {
    ...helveticaNeueLight45,
    fontSize: '1.75rem',
    lineHeight: '1.29', // 36px
    letterSpacing: '-1.6px',
    ...media.from('md').css({
      ...helveticaNeueThin35,
      fontSize: '2.75rem',
      lineHeight: '1.18',
      letterSpacing: '0',
    }),
    [`${StyledHeadingSup}, sup`]: {
      ...baseSupSubScripts,
      fontSize: '1.25rem',
      top: '-1em',
      ...media.from('md').css({ fontSize: '1.25rem', top: '-1.3em' }),
    },
+   '& > span': { letterSpacing: 'inherit' }
  },
```

This solution **can also be used as a workaround** for Heading:

```jsx
<Heading level="h1">
  <span style={{ letterSpacing: 'inherit' }}>My heading</span> is ready
</Heading>
```

### 2) Apply `letterSpacing: inherit` to reset.js

In our `@tds/core-css-reset` package, we can add `letterSpacing: inherit` to ensure all elements on a page will inherit font styles as well as letter spacing.

**reset.js**
```diff
  'html, body, div, span, applet, ...': {
    border: 0,
    font: 'inherit',
+   letterSpacing: 'inherit',
    fontSize: '100%',
    margin: 0,
    padding: 0,
    verticalAlign: 'baseline',
  },
```

**Comment by agorovyi** [comment]
I've tested the option 2 locally in starter kit and it seems to work well. I'm going to proceed with it.

**Comment by agorovyi** [comment]
After a very thorough review of this issue we are going to patch `Heading` component to allow it's children to inherit `letter-spacing` from a parent.

When looking into this problem we identified that Cart page was using an outdated `css-reset` that doesn't have globally defined `letter-spacing` that was added in `css-reset v1.2.2`. We noticed some inconsistencies in the layout of Global Header which relies on our globally defined `letter-spacing`. Also, the `<Text>` component doesn't have a proper letter-spacing in older `css-reset`.

Recommended steps after this `Heading` patch is released:

1. Upgrade `@tds/core-css-reset` to the latest version (with styled-components)
2. Update application to use styled-components `css-reset`: https://github.com/telus/telus-isomorphic-starter-kit/commit/a3cea2b491179328da49b889e19f7babe03f1f61#diff-d92e88124c184ad6a79e4a973ae9b5ba
3. Upgrade `@tds/core-heading` to the latest version that will have this patch that we are going to release.

I've tested all 3 steps locally on unified-commerce-cart application and all the intended changes are being applied correctly.

cc: @msvivianso
